### ISSUE TITLE
Eclipse 4.25 x86_64 keep crashing when using the JS editor  on MacOS Ventura

### ISSUE THREAD

**Issue by ptrussart** [eclipse-platform/eclipse.platform.swt#480]
Hello,

I have a blocking problem with my eclipse installation, whenever I try select a word in the JS editor, my Eclipse always crash with the attached following error.

When I'm working with java files in the java editor, no problem but as long as I try to work with the JS editor, eclipse crash and close.

I'm using Eclipse 4.25 x86_64 / MacOs Ventura /  OpenJDK 11.0.16.

I have tried to run Eclipse in Debug mode and, in the .log file here is the last error message before the crash :

```
SyntaxError: Invalid or unexpected token
    at Module._compile (internal/modules/cjs/loader.js:760:23)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:827:10)
    at Module.load (internal/modules/cjs/loader.js:685:32)
    at Function.Module._load (internal/modules/cjs/loader.js:620:12)
    at Module.require (internal/modules/cjs/loader.js:723:19)
    at require (internal/modules/cjs/helpers.js:14:16)
    at module.exports (/Users/my_user/Documents/Development/Workspaces/any_1_16/.metadata/.plugins/org.eclipse.wildwebdeveloper.embedder.node/node-v16.14.2-darwin-x64/lib/node_modules/npm/lib/cli.js:22:15)
    at Object.<anonymous> (/Users/my_user/Documents/Development/Workspaces/any_1_16/.metadata/.plugins/org.eclipse.wildwebdeveloper.embedder.node/node-v16.14.2-darwin-x64/lib/node_modules/npm/bin/npm-cli.js:2:25)
    at Module._compile (internal/modules/cjs/loader.js:816:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:827:10
```

If I launch Eclipse in command line, when it crashes, I have the following error : 

```
Uncaught exception received.
Error: Connection is closed.
    at se (/Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclipse.wildwebdeveloper_0.6.0.202206151321/node_modules/eslint-server/out/eslintServer.js:1:33782)
    at Object.sendNotification (/Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclipse.wildwebdeveloper_0.6.0.202206151321/node_modules/eslint-server/out/eslintServer.js:1:34678)
    at m.send (/Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclipse.wildwebdeveloper_0.6.0.202206151321/node_modules/eslint-server/out/eslintServer.js:1:133296)
    at m.error (/Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclipse.wildwebdeveloper_0.6.0.202206151321/node_modules/eslint-server/out/eslintServer.js:1:133096)
    at /Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclipse.wildwebdeveloper_0.6.0.202206151321/node_modules/eslint-server/out/eslintServer.js:1:35381
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
Uncaught exception received.
Error: Connection is closed.
    at se (/Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclipse.wildwebdeveloper_0.6.0.202206151321/node_modules/eslint-server/out/eslintServer.js:1:33782)
    at Object.sendNotification (/Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclipse.wildwebdeveloper_0.6.0.202206151321/node_modules/eslint-server/out/eslintServer.js:1:34678)
    at Object.sendNotification (/Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclipse.wildwebdeveloper_0.6.0.202206151321/node_modules/eslint-server/out/eslintServer.js:1:140489)
    at process.exit (/Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclipse.wildwebdeveloper_0.6.0.202206151321/node_modules/eslint-server/out/eslintServer.js:1:174318)
    at Socket.<anonymous> (/Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclipse.wildwebdeveloper_0.6.0.202206151321/node_modules/eslint-server/out/eslintServer.js:1:155291)
    at Socket.emit (node:events:538:35)
    at Pipe.<anonymous> (node:net:687:12)
```

**Comment by akurtakov** [comment]
Most probably duplicate of https://github.com/eclipse-platform/eclipse.platform.swt/issues/452

**Comment by ptrussart** [comment]
It could be linked but I did not use the browser in my case, only the JS editor.

**Comment by Phillipus** [comment]
It will be the cause. See https://github.com/eclipse/wildwebdeveloper/issues/940

**Comment by ptrussart** [comment]
Is there a known workaroud ?

**Comment by Phillipus** [comment]
> Is there a known workaroud ?

If you run Eclipse from the command line. See https://github.com/eclipse-platform/eclipse.platform.swt/issues/452#issuecomment-1295806955

**Comment by florianstephan01** [comment]
I can not confirm that the command line workaround helps. The same issue happens.

**Comment by groumfice** [comment]
Hello,

Same problem Eclipse crashed when open .js files

**Comment by Phillipus** [comment]
This is fixed in macOS Ventura 13.2.

**Comment by groumfice** [comment]
> This is fixed in macOS Ventura 13.2.

I confirm. Seems working now...
Thank's to Apple teams... 👍

**Comment by Phillipus** [comment]
> Thank's to Apple teams...

It was the devs at Webkit who fixed it - https://bugs.webkit.org/show_bug.cgi?id=247387

**Comment by groumfice** [comment]
> > Thank's to Apple teams...
> 
> It was the devs at Webkit who fixed it - https://bugs.webkit.org/show_bug.cgi?id=247387

Ho! A big thank's to us !

**Comment by jmini** [comment]
I had a reproducer in the pom Editor…


Issue with macOS Ventura (13.0.1):

https://user-images.githubusercontent.com/1222165/219305558-24714e2e-84f4-4bdb-af9e-f4e596b75aa3.mov

---

Solved with macOS Ventura (13.2.1):

https://user-images.githubusercontent.com/1222165/219305587-7648d007-1ebe-4e3e-bc3e-289b0e331860.mov


---

And indeed it was crashing when trying to open the preview window of the selected proposal ("Insert comment" in the example). This seems to use Webkit to render some HTML.
